doc: Adding guidance on pinning provider version in our index.md docs (#3640)
mongodb/terraform-provider-mongodbatlas: docs/index.md

}
# Create the resources
```

> 

Reviewer: For best practices, we can also include the point about TF not complaining about non-existent nested attributes, something like 
```
If your config includes additional non-existent attributes within nested structures, Terraform may NOT throw an explicit error even if these attributes are left in the configuration. This is a [known Terraform issue](https://github.com/hashicorp/terraform-plugin-framework/issues/1210). Users should ensure to remove any such attributes from the configuration to avoid any confusion.
```

### Provider and terraform version constraints

> We recommend pinning the [provider version](https://developer.hashicorp.com/terraform/language/providers/requirements#version) of the Atlas Provider to at least the [major version](#versioning-strategy) (e.g. `~> 2.0`) to avoid accidental upgrades to incompatible new versions. From version `2.0.0` onwards our [MongoDB Atlas Provider Versioning Policy](#mongodb-atlas-provider-versioning-policy) ensures minor and patch versions do not include [Breaking Changes](#definition-of-breaking-changes). 

Reviewer 2: This block is too difficult to read, I suggest: 

```suggestion
We recommend that you pin your Atlas [provider version](https://developer.hashicorp.com/terraform/language/providers/requirements#version) to at latest [major version](#versioning-strategy) to avoid accidental upgrades to incompatible new versions. Starting on version `2.0.0`, the [MongoDB Atlas Provider Versioning Policy](#mongodb-atlas-provider-versioning-policy) ensures that minor and patch versions do not include [Breaking Changes](#definition-of-breaking-changes). 
```

> Regarding the terraform version we recommend using the latest [HashiCorp Terraform Core Version](https://github.com/hashicorp/terraform). For more details see [HashiCorp Terraform Version Compatibility Matrix](#hashicorp-terraform-version-compatibility-matrix).

Reviewer 2: What about: 
```suggestion
For Terraform version, we recommend that you use the latest [HashiCorp Terraform Core Version](https://github.com/hashicorp/terraform). For more details see [HashiCorp Terraform Version Compatibility Matrix](#hashicorp-terraform-version-compatibility-matrix).
```
